components/style/gecko/wrapper.rs

```diff
@@ -319,14 +333,23 @@ impl<'ln> TNode for GeckoNode<'ln> {
         panic!("Atomic child count not implemented in Gecko");
     }
 
-    #[inline(always)]
-    fn borrow_data(&self) -> Option<AtomicRef<PersistentStyleData>> {
-        self.get_node_data().as_ref().map(|d| d.0.borrow())
+    fn get_existing_style(&self) -> Option<Arc<ComputedValues>> {
```

Comment:
Note that this is also subtly changing the behavior of most of the other code that used something like `borrow_data().style.as_ref.unwrap()`. Probably you were also aware of it.